calculation/sha_calc/sha_calc/directivity/bea20/distributions.py

```diff
@@ -0,0 +1,314 @@
+from typing import List
+
+import numpy as np
+from scipy import stats
+import matplotlib.pyplot as plt
+from sha_calc.directivity.bea20.EventType import EventType
+
+
+class weibull_truncated_strike_slip(stats.rv_continuous):
+    """Truncates the weibull distribution to between 0 and 1 for strike slip events"""
+
+    weibull = stats.weibull_min(scale=0.626, c=3.921)
+    divide = weibull.cdf(1)
+
+    def _pdf(self, x):
+        return self.weibull.pdf(x) / self.divide
+
+
+class weibull_truncated_oblique(stats.rv_continuous):
+    """Truncates the weibull distribution to between 0 and 1 for oblique events"""
+
+    weibull = stats.weibull_min(scale=0.612, c=3.353)
+    divide = weibull.cdf(1)
+
+    def _pdf(self, x):
+        return self.weibull.pdf(x) / self.divide
+
+
+class gamma_truncated(stats.rv_continuous):
+    """Truncates the gamma distribution to between 0 and 1 for dip slip events"""
+
+    gamma = stats.gamma(a=7.364, scale=0.072)
+    divide = gamma.cdf(1)
+
+    def _pdf(self, x):
+        return self.gamma.pdf(x) / self.divide
+
+
+def monte_carlo_distribution(
```

Comment:
Maybe this and the below function are better off in a different module. Just because they aren't strictly distributions.

Reply:
I thought of `distributions` as different methods of distributing the hypocentres across strike and dip not directly distributions in mathematics, maybe a better name for the file but I can't think of one on the top of my head.

Reply:
Fair, I guess it is under the directivity module. Maybe hypocentre_distributions but that is pretty wordy.